test/run-webhook.py

```diff
@@ -0,0 +1,331 @@
+    print("Get Akri Helm chart: {}".format(helm_chart_location))
+
+    cri_args = shared_test_code.get_cri_args()
+    print("Providing Akri Helm chart with CRI args: {}".format(cri_args))
+
+    extra_helm_args = shared_test_code.get_extra_helm_args()
+    print("Providing Akri Helm chart with extra helm args: {}".format(
+        extra_helm_args))
+
+    helm_install_command = "\
+    helm install {chart_name} {location} \
+    --namespace={namespace} \
+    --set=agent.allowDebugEcho=true \
+    {webhook_config} \
+    {cri_args} \
+    {helm_args} \
+    --debug\
+    ".format(chart_name=HELM_CHART_NAME,
+             location=helm_chart_location,
+             namespace=NAMESPACE,
+             webhook_config=get_webhook_helm_config(),
+             cri_args=cri_args,
+             helm_args=extra_helm_args)
+    print("Helm command: {}".format(helm_install_command))
+    os.system(helm_install_command)
+
+    res = False
+    try:
+        res = do_test()
+    except Exception as e:
+        print(e)
+        res = False
+    finally:
+        # Best effort cleanup work
+        try:
+            # Save Agent and controller logs
+            shared_test_code.save_agent_and_controller_logs(
+                namespace=NAMESPACE)
+        finally:
+            # Delete akri and check that controller and Agent pods deleted
+            os.system("\
+                helm delete {chart_name} \
+                --namespace={namespace}\
+                ".format(
+                chart_name=HELM_CHART_NAME,
+                namespace=NAMESPACE,
+            ))
+            # Delete Webhook Secret
+            client.CoreV1Api().delete_namespaced_secret(name=WEBHOOK_NAME,
+                                                        namespace=NAMESPACE)
+            if res:
+                # Only test cleanup if the test has succeeded up to now
+                if not shared_test_code.check_akri_state(
+                        0, 0, 0, 0, 0, 0, namespace=NAMESPACE):
+                    print(
+                        "Akri not running in expected state after helm delete")
+                    raise RuntimeError("Scenario Failed")
+
+    if not res:
+        raise RuntimeError("Scenario Failed")
+
+
+def do_test() -> bool:
+    kubeconfig_path = shared_test_code.get_kubeconfig_path()
+    print("Loading k8s config: {}".format(kubeconfig_path))
+    config.load_kube_config(config_file=kubeconfig_path)
+
+    # Get kubectl command
+    kubectl_cmd = shared_test_code.get_kubectl_command()
+
+    # Ensure Helm Akri installation applied CRDs and set up Agent and Controller
+    print("Checking for CRDs")
+    if not shared_test_code.crds_applied():
+        print("CRDs not applied by helm chart")
+        return False
+
+    print("Checking for initial Akri state")
+
+    if not shared_test_code.check_akri_state(1, 1, 0, 0, 0, 0):
+        print("Akri not running in expected state")
+        run("sudo {kubectl} get pods,services,akric,akrii --show-labels".
```

Comment:
Curious, how come helm commands do not require sudo, but kubectl do? If possible, would rather not call sudo from within the script so as not to result in a prompt.

Reply:
this is following the example of the existing end-to-end test code.  i'm not sure i can answer the why, but from trial-and-error, sudo seemed to be needed for kubectl.

maybe we need to call the python test scripts with a sudo command so that the scripts themselves don't need to.  maybe this can be part of the test cleanup issue: https://github.com/deislabs/akri/issues/240